common/djangoapps/student/forms.py

```diff
@@ -188,6 +188,19 @@ def __init__(
                                 "required": _("To enroll, you must follow the honor code.")
                             }
                         )
+                elif field_name == 'data_sharing_consent':
+                    if field_value == "required":
+                        self.fields[field_name] = TrueField(
+                            error_messages={
+                                "required": _(
+                                    "You must consent to data sharing to register."
```

Comment:
Thinking this should be reworded as follows:

"To register, you must consent to the edX data sharing policy."

...although I am not sure if we will be publishing a data sharing policy?  (we should, imho)

@edx/doc and/or @griffresch, comments?